integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java

```diff
@@ -137,4 +143,20 @@ public void createFilesTest() throws IOException, TException {
       Assert.assertEquals(TachyonURI.SEPARATOR + k, files.get(k));
     }
   }
+
+  @Test
+  public void workerReRegisterTest() throws Exception {
+    Assert.assertEquals(WORKER_CAPACITY_BYTES, TachyonBlockStore.get().getCapacityBytes());
+
+    List<Pair<Long, TachyonURI>> emptyAnswer = Lists.newArrayList();
+    for (int kills = 0; kills < MASTERS - 1; kills ++) {
+      killLeaderAndWait();
+
+      // TODO(cc) Why this test fail without this line? [TACHYON-970]
+      faultTestDataCheck(emptyAnswer);
```

Comment:
Does the faultTest() work without doing the check? Or is this issue just for this re-register test?